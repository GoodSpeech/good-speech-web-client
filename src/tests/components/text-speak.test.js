import TextSpeak from '../../components/text-speak';
import { render } from '../test-utils';

const defaultProps = {
  text: 'The cat, is under. The table; he said',
  style: {
    color: 'red'
  },
  lang: 'es-AR',
  speechSynthesis: {
    speak: () => {}
  },
  onHover: () => {}
};

describe('TextSpeak component', () => {
  it('Should display text sentences', () => {
    const component = render(TextSpeak, defaultProps);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Should apply hover style when hover a sentence', () => {
    const stopPropagation = jest.fn();
    const preventDefault = jest.fn();
    const component = render(TextSpeak, defaultProps);
    const sentence = component.toJSON().children[0];
    
    sentence.props.onMouseEnter({stopPropagation, preventDefault});
    expect(component.toJSON()).toMatchSnapshot();
    sentence.props.onMouseLeave({stopPropagation, preventDefault});
    expect(component.toJSON()).toMatchSnapshot();

    expect(stopPropagation.mock.calls.length).toBe(2);
    expect(preventDefault.mock.calls.length).toBe(2);
  });

  it('Should apply hover style when hover a sentence', () => {
    const stopPropagation = jest.fn();
    const preventDefault = jest.fn();
    const component = render(TextSpeak, defaultProps);
    const sentence = component.toJSON().children[0];
    
    sentence.props.onMouseDown({stopPropagation, preventDefault});
    expect(component.toJSON()).toMatchSnapshot();

    expect(stopPropagation).toBeCalled();
    expect(preventDefault).toBeCalled();
  });

  it('Should call speechSynthesis.speak(text) when click a sentence', () => {
    const speak = jest.fn();
    const component = render(TextSpeak, defaultProps, {
      speechSynthesis: { speak }
    });
    const sentence = component.toJSON().children[0];
    
    sentence.props.onClick();
    expect(component.toJSON()).toMatchSnapshot();

    expect(speak).toBeCalledWith('The cat,', 'es-AR');
  });

});