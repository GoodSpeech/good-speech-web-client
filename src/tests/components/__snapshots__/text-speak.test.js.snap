// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextSpeak component Should apply hover style when hover a sentence 1`] = `
<span>
  <span
    className={undefined}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "borderRadius": 3,
        "boxShadow": "inset 0 0 2px 2px #ccc",
        "color": "red",
        "cursor": "pointer",
      }
    }
  >
    The cat,
  </span>
  <span
    className={undefined}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
  >
     is under.
  </span>
  <span
    className={undefined}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
  >
     The table;
  </span>
  <span
    className={undefined}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
  >
     he said
  </span>
</span>
`;

exports[`TextSpeak component Should apply hover style when hover a sentence 2`] = `
<span>
  <span
    className={undefined}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
  >
    The cat,
  </span>
  <span
    className={undefined}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
  >
     is under.
  </span>
  <span
    className={undefined}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
  >
     The table;
  </span>
  <span
    className={undefined}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
  >
     he said
  </span>
</span>
`;

exports[`TextSpeak component Should apply hover style when hover a sentence 3`] = `
<span>
  <span
    className={undefined}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
  >
    The cat,
  </span>
  <span
    className={undefined}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
  >
     is under.
  </span>
  <span
    className={undefined}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
  >
     The table;
  </span>
  <span
    className={undefined}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
  >
     he said
  </span>
</span>
`;

exports[`TextSpeak component Should call speechSynthesis.speak(text) when click a sentence 1`] = `
<span>
  <span
    className={undefined}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
  >
    The cat,
  </span>
  <span
    className={undefined}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
  >
     is under.
  </span>
  <span
    className={undefined}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
  >
     The table;
  </span>
  <span
    className={undefined}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
  >
     he said
  </span>
</span>
`;

exports[`TextSpeak component Should display text sentences 1`] = `
<span>
  <span
    className={undefined}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
  >
    The cat,
  </span>
  <span
    className={undefined}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
  >
     is under.
  </span>
  <span
    className={undefined}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
  >
     The table;
  </span>
  <span
    className={undefined}
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
  >
     he said
  </span>
</span>
`;
